### PR TITLE
Modernize `TrackSplittingMonitor` and fix bug with filling ME-s

### PR DESCRIPTION
**Pull request by mmusich**
#### PR description:

The main goal of this PR is to fix the filling of the residuals in the `TrackSplittingMonitor` DQM code. I noticed that in recent cosmic data these plots are almost all empty (see e.g.https://tinyurl.com/2q9tgomk). This was caused by a bug in the filling of the `MonitorElements` which is fixed in this PR.
I profit of it to:
   * modernize the code of the module
   * introduce a `fillDescriptions` method and use it in the related configurations
   * remove the `TrackSplitMonitor` from the Tracking DQM collisions setup as anyway there are no right input tracks to be fed to the module.

#### PR validation:

Run successfully `runTheMatrix.py -l 138.1 -t 4 -j 8 --custom_command='-n 5000' --ibeos` and checked that the cosmic plots are all filled.

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

Not a backport, but needs to be backported to CMSSW_13_0_X for 2023 data-takging purposes.

### CODE REVIEW

**Comment by mmusich** [comment]
type bug-fix

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by emanueleusai** [comment]
+1

File: DQM/TrackingMonitor/src/TrackSplittingMonitor.cc [hunk 1]
```diff
@@ -241,7 +230,8 @@ void TrackSplittingMonitor::analyze(const edm::Event& iEvent, const edm::EventSe
     if ((nRechitinBPIX1 >= pixelHitsPerLeg_) && (nRechitinBPIX1 >= pixelHitsPerLeg_) &&
         (nRechits1 >= totalHitsPerLeg_) && (nRechits2 >= totalHitsPerLeg_)) {
       // dca cut
-      if (((fabs(d01) < d0Cut_)) && (fabs(d02) < d0Cut_) && (fabs(dz2) < dzCut_) && (fabs(dz2) < dzCut_)) {
+      if (((std::abs(d01) < d0Cut_)) && (std::abs(d02) < d0Cut_) && (std::abs(dz2) < dzCut_) &&
+          (std::abs(dz2) < dzCut_)) {
```
Comment (perrotta) [inline DQM/TrackingMonitor/src/TrackSplittingMonitor.cc:234]: Shouldn't one of the `dz2` be a `dz1` instead?
```suggestion
      if (((std::abs(d01) < d0Cut_)) && (std::abs(d02) < d0Cut_) && (std::abs(dz1) < dzCut_) &&
          (std::abs(dz2) < dzCut_)) {
```

File: DQM/TrackingMonitorSource/python/TrackingSourceConfig_Tier0_cff.py [hunk 2]
```diff
@@ -29,9 +29,9 @@
 #TrackEffMon_bhmuon.AlgoName                        = 'BHMuonTk'
 #TrackEffMon_bhmuon.FolderName                      = 'Tracking/TrackParameters/TrackEfficiency'
 
-# Split Tracking
-from  DQM.TrackingMonitor.TrackSplittingMonitor_cfi import *
-TrackSplitMonitor.FolderName = 'Tracking/TrackParameters/SplitTracks'
+# # Split Tracking
```
Comment (perrotta) [inline DQM/TrackingMonitorSource/python/TrackingSourceConfig_Tier0_cff.py:32]:
```suggestion
# Split Tracking
```
(not important, but if you have to modify anyhow...)

File: DQM/TrackingMonitor/src/TrackSplittingMonitor.cc (hunk 1 above)
Comment (mmusich) [reply DQM/TrackingMonitor/src/TrackSplittingMonitor.cc:234]: was like that even before.

File: DQM/TrackingMonitorSource/python/TrackingSourceConfig_Tier0_cff.py (hunk 2 above)
Comment (mmusich) [reply DQM/TrackingMonitorSource/python/TrackingSourceConfig_Tier0_cff.py:32]: not sure to see why that would be needed.

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by mmusich** [comment]
@emanueleusai @perrotta PR is ready for further review.

**Comment by perrotta** [comment]
> @emanueleusai @perrotta PR is ready for further review.

This is fine with me, I did not spot other issues besides that (pre-existent) bug to be fixed

**Comment by emanueleusai** [comment]
+1

**Comment by perrotta** [comment]
+1

**Pull request merged**